test/dynamo/test_aot_autograd.py

```diff
@@ -332,6 +332,58 @@ def guard_fail_fn(failure):
         self.assertEqual(cc.frame_count, 1)
         self.assertTrue(failure_reason is None)
 
+    def test_double_backward_errors(self):
+        # Remove this test after we get double backward to actually work
+        for grad_output in (torch.tensor(1.0, requires_grad=True), None):
+            # See @once_differentiable docs for why there are two different errors
+            x = torch.tensor(1.0, requires_grad=True)
+            if grad_output is None:
+                err = "does not require grad and does not have a grad_fn"
```

Comment:
This error message isn't sufficient because autograd-related subsystems have different behavior depending on if the outputs are not related to the input -- we should always try to raise an error about torch.compile. Here's an example:

Using a strategy like torch.autograd.functional.hessian to compute hessians (functorch.hessian has a similar strategy but also gets a much different error message on this case), we get zero hessians for a once_differentiable function. This would be silently incorrect -- we don't want to give zero second derivative for a torch.compile'd function in any case, we want to give an error message.
```py
import torch

class Func(torch.autograd.Function):
    def forward(ctx, x):
        ctx.save_for_backward(x)
        return x.sin()

    @torch.autograd.function.once_differentiable
    def backward(ctx, gy):
        x, = ctx.saved_tensors
        return gy * x.cos()

x = torch.tensor(1., requires_grad=True)
ggx = torch.autograd.functional.hessian(Func.apply, x)
```

Reply:
If we can't find a better solution than `@once_differentiable` then I'm fine with the current solution in this PR as a stopgap, because it is less silently incorrect on fewer cases than before.

Reply:
I think we can improve on `@once_differentiable` if we specialize for this specific case. Updated now.